tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/CosmosDatabaseInitializer.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+using Azure.Sdk.Tools.PipelineWitness.Configuration;
+
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Azure.Sdk.Tools.PipelineWitness
+{
+    internal class CosmosDatabaseInitializer : IHostedService
+    {
+        private readonly CosmosClient cosmosClient;
+        private readonly IOptions<PipelineWitnessSettings> options;
+
+        public CosmosDatabaseInitializer(CosmosClient cosmosClient, IOptions<PipelineWitnessSettings> options)
+        {
+            this.cosmosClient = cosmosClient;
+            this.options = options;
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            var settings = this.options.Value;
+
+            Database database = await this.cosmosClient.CreateDatabaseIfNotExistsAsync(settings.CosmosDatabase, cancellationToken: cancellationToken);
```

Comment:
If you spin up multiple pipeline witness instances, is this going to be a safe concurrent operation? Or would most instances just fail and restart?

Reply:
I believe concurrent CreateIfNotExist calls result in success because they eat conflicts as proof of existence.

In a perfect world, the database and container preexist the app and the app uses a database level principal with only document crud access.

Reply:
Makes sense, I would say we should have a follow-up to move the database initialization into the buildout bicep templates so the principal permissions can be limited as you suggest.